math/mathmore/test/StatFunction.h

```diff
@@ -0,0 +1,107 @@
+// @(#)root/test:$Id$
+// Author: Lorenzo Moneta       06/2005
+//         Martin Storø Nyfløtt 05/2017
+
+#ifndef STATFUNCTION
+#define STATFUNCTION
+
+#include "Math/AllIntegrationTypes.h"
+#include "Math/IParamFunction.h"
+#include "Math/RootFinderAlgorithms.h"
+#include "Math/RootFinder.h"
+
+#include <iomanip>
+
+using namespace ROOT::Math;
+
+#define INF std::numeric_limits<double>::infinity()
+
+// typedef for a free function like gamma(double x, double a, double b)
+
+typedef std::function<double(double, double, double)> AlgoFunc;
+
+// statistical function class
+const int N_PAR = 2;
+
+class StatFunction : public ROOT::Math::IParamFunction {
+private:
+   double DoEvalPar(double x, const double *) const
+   {
+      // use explicitly cached param values
+      return fPdf(x, fParams[0], fParams[1]);
+   }
+
+   AlgoFunc fPdf;
+   AlgoFunc fCdf;
+   AlgoFunc fQuant;
+   double fParams[N_PAR];
+   double fScaleIg;
+   double fScaleDer;
+   double fScaleInv;
+   int NFuncTest;
+   double xmin;
```

Comment:
all data member must start with the letter f followed by CamelCase words.  Thanks.